### PR TITLE
[MM-39827] Remove AdminAccessToken & PermissionActAsAdmin

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
As discussed offline, the `AdminAccessToken` has no clear use case and caused confusion.  This  PR removes it alongside the  `PermissionActAsAdmin`.

This PR targets https://github.com/mattermost/mattermost-plugin-apps/pull/278 to avoid merge conflicts. It should be merged after https://github.com/mattermost/mattermost-plugin-apps/pull/278.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-39827

### Breaking Changes
- Removed `AdminAccessToken` from `Expand`
- Removed `AdminAccessToken` from `Call.Context`
- Removed `PermissionActAsAdmin` permission

The following apps need to get updated:
- [x] Examples
- [x] Test app, https://github.com/mattermost/mattermost-app-test/pull/11
- [x] ~~ServiceNow~~
- [x] ~~Zendesk~~

### CODE REVIEW

**Comment by hanzei** [comment]
Force pushed as the history was messed up.

**Comment by hanzei** [comment]
CI will fail until https://github.com/mattermost/mattermost-plugin-apps/pull/280 is merged

**Review by catalintomai — APPROVED** [review]

File: server/builtin/debug_clean.go [hunk 1]
```diff
@@ -28,8 +28,7 @@ func (a *builtinApp) debugClean() handler {
 				Call: &apps.Call{
 					Path: pDebugClean,
 					Expand: &apps.Expand{
-						AdminAccessToken: apps.ExpandAll, 
-						Locale:           apps.ExpandAll,
+						Locale: apps.ExpandAll,
```
Comment (levb) [inline server/builtin/debug_clean.go:31]: I am confused as to how is the enforcement of admin-only is done now for debug clean

File: server/proxy/incoming.go [hunk 2]
```diff
@@ -110,23 +107,13 @@ func (in *Incoming) ensureUserToken(mm *pluginapi.Client) error {
 	return nil
 }
 
-func (p *Proxy) getAdminClient(in Incoming) (mmclient.Client, error) {
+func (p *Proxy) getClient(in Incoming) (mmclient.Client, error) {
```
Comment (levb) [inline server/proxy/incoming.go:110]: 1/5 `getUserClient` (as opposed to Bot?)

File: server/builtin/debug_clean.go (hunk 1 above)
Comment (levb) [reply server/builtin/debug_clean.go:31]: I think just need to add `ActingUser: expand.Summary,`

**Review by levb — APPROVED** [review]

**Pull request merged**